subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesGroovyDSLDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,1930 @@
+            package org.sample;
+
+            import org.apache.commons.beanutils.PropertyUtils;
+
+            public class PersonTest {
+                @Test
+                public void testPerson() {
+                    Object person = new Person();
+                    PropertyUtils.setSimpleProperty(person, "name", "Bart Simpson");
+                    PropertyUtils.setSimpleProperty(person, "age", 38);
+                    assertEquals("Bart Simpson", person.getName());
+                    assertEquals(38, person.getAge());
+                }
+            }
+        """
+
+        expect:
+        succeeds 'checkConfiguration'
+
+        where:
+        desc                | dependencyNotation
+        'GAV string'        | gavStr(beanUtilsGroup, beanUtilsName, beanUtilsVer)
+        'GAV map'           | gavMap(beanUtilsGroup, beanUtilsName, beanUtilsVer)
+        'named args'        | namedArgs(beanUtilsGroup, beanUtilsName, beanUtilsVer)
+    }
+
+    def "can add dependencies using a non-String CharSequence: #type"() {
+        given:
+        buildFile << """
+        import org.apache.commons.lang3.text.StrBuilder;
+
+        buildscript {
+            ${mavenCentralRepository()}
+
+            dependencies {
+                classpath("org.apache.commons:commons-lang3:3.11")
+            }
+        }
+
+        plugins {
+          id 'java-library'
+        }
+
+        ${mavenCentralRepository()}
+
+        $type buf = $creationNotation
+
+        testing {
+            suites {
+                test {
+                    dependencies {
+                        implementation(buf)
+                    }
+                }
+            }
+        }
+
+        tasks.register('checkConfiguration') {
+            dependsOn test
+            doLast {
+                def testCompileClasspathFileNames = configurations.testCompileClasspath.files*.name
+                assert testCompileClasspathFileNames.containsAll('commons-lang3-3.11.jar')
+            }
+        }
+        """
+
+        expect:
+        succeeds 'checkConfiguration'
+
+        where:
+        type            | creationNotation
+        'StringBuffer'  | "new StringBuffer('org.apache.commons:commons-lang3:3.11')"
+        'StringBuilder' | "new StringBuilder('org.apache.commons:commons-lang3:3.11')"
+        'StrBuilder'    | "new StrBuilder('org.apache.commons:commons-lang3:3.11')"
+        'GString'       | '"org.apache.commons:commons-lang3:3.${(11 + 11) / (2 + 1 - 1)}"'
+    }
+
+    def "can NOT add a list of GAV dependencies to #suiteDesc"() {
+        given:
+        buildFile << """
+            plugins {
+              id 'java-library'
+            }
+
+            ${mavenCentralRepository()}
+
+
+            testing {
+                suites {
+                    $suiteDeclaration {
+                        dependencies {
+                            implementation(['org.apache.commons:commons-lang3:3.11', 'com.google.guava:guava:30.1.1-jre'])
```

Comment:
It looks like this works in Groovy:
```
   implementation 'org.apache.commons:commons-text', 'org.apache.commons:commons-lang3'
```
but this does not:
```
   implementation ['org.apache.commons:commons-text', 'org.apache.commons:commons-lang3']
```
in the top-level dependencies block.

I think we should check for the first one, which means dropping the `[]`'s

Reply:
Expanded this test to demonstate working and failing syntax in each DSL.